pkg/health/health_manager.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/cilium/cilium/pkg/health/defaults"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/mtu"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/pidfile"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const (
+	controllerInterval    = 60 * time.Second
+	successfulPingTimeout = 3 * time.Minute
+)
+
+// Cell provides the Cilium health infrastructure that is responsible for
+// checking the connectivity between Cilium nodes and Cilium endpoints.
+var Cell = cell.Module(
+	"cilium-health",
+	"Cilium health infrastructure",
+	cell.Provide(newHealthProvider),
+)
+
+type CiliumHealthManager interface {
+	Init(ctx context.Context, routingInfo *linuxrouting.RoutingInfo) error
+	GetStatus() *models.Status
+}
+
+type ciliumHealthManager struct {
+	logger          *slog.Logger
+	healthSpec      *healthApi.Spec
+	sysctl          sysctl.Sysctl
+	loader          datapath.Loader
+	mtuConfig       mtu.MTU
+	bigTCPConfig    *bigtcp.Configuration
+	endpointCreator endpointcreator.EndpointCreator
+	endpointManager endpointmanager.EndpointManager
+	k8sClientSet    k8sClient.Clientset
+
+	ciliumHealth *CiliumHealth
+}
+
+type ciliumHealthParams struct {
+	cell.In
+
+	Logger          *slog.Logger
+	Lifecycle       cell.Lifecycle
+	HealthSpec      *healthApi.Spec
+	Sysctl          sysctl.Sysctl
+	Loader          datapath.Loader
+	MtuConfig       mtu.MTU
+	BigTCPConfig    *bigtcp.Configuration
+	EndpointCreator endpointcreator.EndpointCreator
+	EndpointManager endpointmanager.EndpointManager
+	K8sClientSet    k8sClient.Clientset
+}
+
+func newHealthProvider(params ciliumHealthParams) CiliumHealthManager {
+	h := &ciliumHealthManager{
+		logger:          params.Logger,
+		healthSpec:      params.HealthSpec,
+		sysctl:          params.Sysctl,
+		loader:          params.Loader,
+		mtuConfig:       params.MtuConfig,
+		bigTCPConfig:    params.BigTCPConfig,
+		endpointCreator: params.EndpointCreator,
+		endpointManager: params.EndpointManager,
+		k8sClientSet:    params.K8sClientSet,
+	}
+
+	params.Lifecycle.Append(cell.Hook{
+		OnStop: func(hookContext cell.HookContext) error {
+			// Make sure to clean up the endpoint namespace when cilium-agent terminates
+			h.cleanupHealthEndpoint()
+			return nil
+		},
+	})
+
+	return h
+}
+
+func (h *ciliumHealthManager) Init(ctx context.Context, routingInfo *linuxrouting.RoutingInfo) error {
+	// Launch cilium-health in the same process (and namespace) as cilium.
+	h.logger.Info("Launching Cilium health daemon")
+	ch, err := h.launchCiliumNodeHealth(h.healthSpec, h.loader.HostDatapathInitialized())
+	if err != nil {
+		return fmt.Errorf("failed to start cilium health: %w", err)
+	}
+
+	h.ciliumHealth = ch
+
+	// If endpoint health checking is disabled, the virtual endpoint does not need to be launched
+	if !option.Config.EnableEndpointHealthChecking {
```

Comment:
Can we make this a config cell?

Reply:
unfortunately this config value is also used in other places :disappointed: